src/hooks/useProductQuery.tsx

```diff
@@ -0,0 +1,12 @@
+import { useQuery } from 'react-query'
+import ky from 'ky'
+
+function useProductQuery(productId: string) {
+  return useQuery('product', async () => {
+    const response = ky.get(`https://fakestoreapi.com/products/${productId}`)
+    const data: ProductItem = await response.json()
+    return data
+  })
```

Comment:
To make sure we only call the query when productId is defined `useQuery` as a super helpful `enabled` property
```suggestion
  }, { enabled: !!productId })
```